repo-agent/review-ui/ui/src/App.js

```diff
@@ -431,6 +437,29 @@ function App() {
       .catch(err => console.error("Failed to delete issue:", err));
   };
 
+  const handleMoveCommentAndSave = (id, index, newPath, newLine, newSide) => {
+    setDrafts(prevDrafts => {
+      const newDrafts = JSON.parse(JSON.stringify(prevDrafts));
+      const draftToUpdate = newDrafts[id];
+      if (draftToUpdate && draftToUpdate.review && draftToUpdate.review.comments && draftToUpdate.review.comments[index]) {
+        const commentToUpdate = draftToUpdate.review.comments[index];
+        commentToUpdate.path = newPath;
+        commentToUpdate.line = newLine;
+        commentToUpdate.side = newSide;
+
+        const draftYaml = yaml.dump(draftToUpdate);
+        fetch(`/api/repo/${activeRepo.namespace}/${activeRepo.name}/prs/${id}/draft`, {
+          method: 'POST',
+          headers: { 'Content-Type': 'application/json' },
+          body: JSON.stringify({ draft: draftYaml })
+        }).catch(err => console.error("Failed to save draft:", err));
```

Comment:
nit: currently if we fail to save the draft the UI is still updated so the draft and the UI are out of sync.  In this case it make sense to revert back to the prior state of the saved draft

Reply:
i honestly dont know how to fix this. Also how do i verify it. Lets get it in and treat it as a bug later on.